fault-proof/src/proposer.rs

```diff
@@ -138,6 +138,14 @@ impl ProposerState {
     }
 }
 
+/// Snapshot of the proposer's cached state for testing and monitoring.
+#[derive(Clone, Debug)]
+pub struct ProposerStateSnapshot {
+    pub anchor_index: Option<U256>,
```

Comment:
This is exactly what I mentioned today. The system only cares about the anchor index!

Reply:
Yeah I found that too! Do you want me to include the refacor in this PR? Or should we keep it in a separate ticket?